drivers/include/lm75.h

```diff
@@ -0,0 +1,240 @@
+ *
+ * @author      Vitor Batista <vitor.batista@ml-pa.com>
+ *
+ * @}
+ */
+
+/**
+ * @defgroup    drivers_lm75     LM75 Temperature Sensor driver compile configuration
+ * @ingroup     drivers_sensors
+ * @brief       Driver for the lm75 temperature sensors.
+ */
+
+#ifndef LM75_H
+#define LM75_H
+
+#include "periph/i2c.h"
+#include "periph/gpio.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name LM75 return values
+ */
+enum {
+    LM75_SUCCESS = 0,
+    LM75_ERROR_I2C,
+    LM75_ERROR
+};
+
+/**
+ * @brief temperature reading properties/resolutions struct of the LM75 sensors
+ */
+typedef struct lm75_properties {
+    uint16_t    os_res;      /**< resolution of the OS and HYST registers */
+    uint16_t    os_mult;     /**< multiplier required for getting the OS and HYST into ºC */
+    uint16_t    temp_res;    /**< resolution of the temperature register */
+    uint16_t    temp_mult;   /**< multiplier required for getting the temperature into ºC */
+    uint8_t     os_shift;    /**< how many bits need to be shifted (2 bytes - any unused bits) */
+    uint8_t     temp_shift;  /**< how many bits need to be shifted (2 bytes - any unused bits) */
+} lm75_properties_t;
+
+extern lm75_properties_t lm75a_properties;      /**< declaration present in lm75.c */
+extern lm75_properties_t tmp1075_properties;    /**< declaration present in lm75.c */
+
+/**
+ * @brief params required for initialization
+ */
+typedef struct lm75_params {
+    const lm75_properties_t *res;         /**< Temperature resolutions */
+    uint16_t conv_rate;                   /**< Conversion rate in ms */
+    i2c_t    i2c_bus;                     /**< I2C Bus used */
+    uint8_t  i2c_addr;                    /**< i2c address */
+    uint8_t  shutdown_mode;               /**< Shutdown mode register */
+    uint8_t  tm_mode;                     /**< Thermistor Mode */
+    uint8_t  polarity;                    /**< OS polarity register */
+    uint8_t  fault_q;                     /**< Fault Queue register */
+    /* only configurable for the TMP1075 */
+    uint8_t  conv_rate_reg;               /**< Device Conversion rate register */
+} lm75_params_t;
+
+/**
+ * @brief lm75 device descriptor
+ */
+typedef struct lm75 {
+    lm75_params_t    lm75_params; /**< Configuration Parameters */
+} lm75_t;
+
+/**
+ * @brief Initialization of the LM75 sensor
+ *
+ * Initializes the sensor according to specific input parameters.
+ *
+ * @param[out] dev        device structure to initialize
+ * @param[in] params      initialization parameters
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ * @return LM75_ERROR, on initialization related error
+ */
+int lm75_init(lm75_t *dev, const lm75_params_t *params);
+
+/**
+ * @brief Temperature values of LM75 sensor
+ *
+ * Reads the sensor temperature values from TEMP_REG
+ * the value is given with the full precision the device is capable of
```

Comment:
It would still be nice to have a description of what one is supposed to do with that data